test/units/TEST-65-ANALYZE.sh

```diff
@@ -172,6 +172,15 @@ systemd-analyze calendar --base-time=yesterday --iterations=5 '*-* *:*:*'
 systemd-analyze timestamp now
 systemd-analyze timestamp -- -1
 systemd-analyze timestamp yesterday now tomorrow
+systemd-analyze timestamp 'Fri 2012-11-23 23:02:15'
+systemd-analyze timestamp 'Fri 2012-11-23 23:02:15 UTC'
+systemd-analyze timestamp 'Fri 2012-11-23 23:02:15 CET'
+for i in $(timedatectl list-timezones); do
+    # FIXME(?): NZ is broken???
+    [[ "$i" == NZ ]] && continue
```

Comment:
this is actually a bug in our parser.

it's the "welded RFC3339" style, that uses "Z" as suffix. i.e. the first if check in parse_timestamp() misinterprets "NZ" suffix as such an ISO timestamp.

I guess the line needs to be changed to:

```c
if (t_len > 2 && t[t_len - 1] == 'Z' && undecchar(t[t_len - 2]) >= 0)  /* RFC3339-style welded UTC: "1985-04-12T23:20:50.52Z" */
```

or so

Reply:
i.e. it's caused by ef658a63f81. Any chance you can add a commit that tweaks this, and mentions it is a follow-up for ef658a63f81?